historical_system_profiles/views/v0.py

```diff
@@ -49,18 +40,51 @@ def _get_current_names_for_profiles(hsps):
     return enriched_hsps
 
 
+def _filter_old_hsps(hsps):
+    """
+    removes any HSPs with a captured_date older than now minus the valid profile age
+    """
+    now = datetime.datetime.now(datetime.timezone.utc)
+    cutoff = now - relativedelta(seconds=config.valid_profile_age_seconds)
+
+    valid_hsps = []
+    for hsp in hsps:
+        captured_date = dateutil.parser.parse(hsp.captured_date)
+        if captured_date > cutoff:
+            valid_hsps.append(hsp)
+
+    return valid_hsps
+
+
+def get_hsps_by_ids(profile_ids):
+    """
+    return a list of historical system profiles for the given profile IDs
+    """
+    account_number = view_helpers.get_account_number(request)
+
+    result = db_interface.get_hsps_by_profile_ids(profile_ids, account_number)
+    filtered_result = _filter_old_hsps(result)
+
+    result_with_updated_names = _get_current_names_for_profiles(filtered_result)
+
+    return {"data": [r.to_json() for r in result_with_updated_names]}
+
+
 def get_hsps_by_inventory_id(inventory_id):
     """
     return a list of historical system profiles for a given inventory id
     """
     account_number = view_helpers.get_account_number(request)
     query_results = db_interface.get_hsps_by_inventory_id(inventory_id, account_number)
+    valid_profiles = _filter_old_hsps(query_results)
 
-    result = {
-        "profiles": [
-            {"captured_date": p.captured_date, "id": p.id} for p in query_results
-        ],
-    }
+    profile_metadata = []
+    for profile in valid_profiles:
+        profile_metadata.append(
+            {"captured_date": profile.captured_date, "id": profile.id}
+        )
+
```

Comment:
no change to logic here really, besides passing in valid_profiles instead of query_results. I rewrote it to avoid the comprehension.